tests/lib/rules/no-unused-prop-types.js

```diff
@@ -1461,6 +1461,168 @@ ruleTester.run('no-unused-prop-types', rule, {
+        '    bar: PropTypes.func,',
+        '    baz: PropTypes.func,',
+        '  }',
+        '  classProperty = async () => {',
+        '    const { foo, bar, baz } = this.props;',
+        '    await foo();',
+        '    await bar();',
+        '    await baz();',
+        '  };',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // Props used inside of an async class method
+      code: [
+        'export class Example extends Component {',
+        '  static propTypes = {',
+        '    foo: PropTypes.func,',
+        '  }',
+        '  async method() {',
+        '    await this.props.foo();',
+        '  };',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // Multiple props used inside of an async class method
+      code: [
+        'export class Example extends Component {',
+        '  static propTypes = {',
+        '    foo: PropTypes.func,',
+        '    bar: PropTypes.func,',
+        '    baz: PropTypes.func,',
+        '  }',
+        '  async method() {',
+        '    await this.props.foo();',
+        '    await this.props.bar();',
+        '    await this.props.baz();',
+        '  };',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // Destrucuted props inside of async class method
+      code: [
+        'export class Example extends Component {',
+        '  static propTypes = {',
+        '    foo: PropTypes.func,',
+        '  }',
+        '  async method() {',
+        '    const { foo } = this.props;',
+        '    await foo();',
+        '  };',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // Multiple destrucuted props inside of async class method
+      code: [
+        'export class Example extends Component {',
+        '  static propTypes = {',
+        '    foo: PropTypes.func,',
+        '    bar: PropTypes.func,',
+        '    baz: PropTypes.func,',
+        '  }',
+        '  async method() {',
+        '    const { foo, bar, baz } = this.props;',
+        '    await foo();',
+        '    await bar();',
+        '    await baz();',
+        '  };',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // factory functions that return async functions
+      code: [
+        'export class Example extends Component {',
+        '  static propTypes = {',
+        '    foo: PropTypes.func,',
+        '    bar: PropTypes.func,',
+        '    baz: PropTypes.func,',
+        '  }',
+        '  factory() {',
+        '    return async () => {',
+        '      await this.props.foo();',
+        '      await this.props.bar();',
+        '      await this.props.baz();',
+        '    };',
+        '  }',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
+    }, {
+      // factory functions that return async functions
+      code: [
+        'export class Example extends Component {',
+        '  static propTypes = {',
```

Comment:
I added a few more cases with async functions defined in the render method rather than returned from a separate method, and used the default eslint parser.

The issue was with marking the props as used, not with how the propTypes were defined, so using OG syntax works just fine.